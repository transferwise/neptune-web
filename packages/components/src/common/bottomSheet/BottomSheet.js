import React, { forwardRef } from 'react';
import Types from 'prop-types';
import classNames from 'classnames';

import Dimmer from '../../dimmer';
import SlidingPanel from '../../slidingPanel';
import './BottomSheet.css';
import { Position } from '../propsValues/position';

const BottomSheet = forwardRef(({ children, open, className }, ref) => (
  <Dimmer open={open}>
    <SlidingPanel
      open={open}
      position={Position.BOTTOM}
      className={classNames('np-bottom-sheet', className)}
      ref={ref}
    >
      {children}
    </SlidingPanel>
  </Dimmer>
));

BottomSheet.propTypes = {
  children: Types.node.isRequired,
  className: Types.string,
  open: Types.bool,
};

BottomSheet.defaultProps = {
  className: undefined,
  open: false,
};

export default BottomSheet;
