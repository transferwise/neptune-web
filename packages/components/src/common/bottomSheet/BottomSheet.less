.np-bottom-sheet {
  box-shadow: 0px 0px 40px rgba(34, 48, 73, 0.2);
  border-radius: var(--radius-medium) var(--radius-medium) 0 0;
}
