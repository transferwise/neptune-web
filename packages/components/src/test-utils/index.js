import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from '..';
import { DEFAULT_LOCALE } from '../common/locale';
import en from '../../i18n/en.json';

/**
 * Custom `render` function which wraps passed elements in Provider component
 * Source: https://testing-library.com/docs/example-react-intl/#creating-a-custom-render-function
 */
function customRender(ui, { locale = DEFAULT_LOCALE, messages = en, ...renderOptions } = {}) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return <Provider i18n={{ locale, messages }}>{children}</Provider>;
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { userEvent };
export { customRender as render };
