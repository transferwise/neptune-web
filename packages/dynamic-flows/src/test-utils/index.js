import React from 'react';
import { mount } from 'enzyme';
import { Provider } from '@transferwise/components';
import { act } from 'react-dom/test-utils';

const getMockFetchPromise = (status, jsonFn, delay, signal) => {
  const response = {
    status,
    json: jsonFn,
  };
  return new Promise((resolve) => {
    let aborted = false;
    if (signal) {
      signal.addEventListener('abort', () => {
        aborted = true;
      });
    }
    setTimeout(() => {
      return !aborted && resolve(response);
    }, delay);
  });
};

const wait = (t) => {
  return act(() => {
    return new Promise((resolve) => setTimeout(resolve, t));
  });
};

function mountWithProviders(component, locale = 'en-GB', messages = {}) {
  return mount(<Provider i18n={{ locale, messages }}>{component}</Provider>);
}

export { mountWithProviders as mount, getMockFetchPromise, wait };
