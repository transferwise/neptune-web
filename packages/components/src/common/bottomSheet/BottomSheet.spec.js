import React from 'react';
import { render } from '../../test-utils';

import BottomSheet from './BottomSheet';

jest.mock('../../dimmer', () => {
  // eslint-disable-next-line react/prop-types
  return ({ open, children }) => {
    return open ? <div className="dimmer">{children}</div> : null;
  };
});

jest.mock('../../slidingPanel', () => {
  // eslint-disable-next-line react/prop-types
  return ({ open, children }) => (open ? <div className="sliding-panel">{children}</div> : null);
});

describe('BottomSheet', () => {
  it('renders content when open', async () => {
    const { container } = render(<BottomSheet open>content</BottomSheet>);
    expect(container).toMatchSnapshot();
  });

  it("doesn't renders content when closed", async () => {
    const { container } = render(<BottomSheet>content</BottomSheet>);

    expect(container).toMatchSnapshot();
  });
});
