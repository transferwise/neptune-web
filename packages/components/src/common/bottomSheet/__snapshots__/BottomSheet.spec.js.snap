// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomSheet doesn't renders content when closed 1`] = `<div />`;

exports[`BottomSheet renders content when open 1`] = `
<div>
  <div
    class="dimmer"
  >
    <div
      class="sliding-panel"
    >
      content
    </div>
  </div>
</div>
`;
